src/IssueViz/FixSuggestion/FixSuggestionHandler.cs

```diff
@@ -19,15 +19,99 @@
+    private readonly IIssueSpanCalculator issueSpanCalculator;
+
+    [ImportingConstructor]
+    internal FixSuggestionHandler(ILogger logger, IDocumentNavigator documentNavigator, IIssueSpanCalculator issueSpanCalculator, IOpenInIdeConfigScopeValidator openInIdeConfigScopeValidator, IIDEWindowService ideWindowService) : 
+        this(ThreadHandling.Instance, logger, documentNavigator, issueSpanCalculator, openInIdeConfigScopeValidator, ideWindowService)
+    {
+    }
+
+    internal FixSuggestionHandler(
+        IThreadHandling threadHandling,
+        ILogger logger,
+        IDocumentNavigator documentNavigator,
+        IIssueSpanCalculator issueSpanCalculator,
+        IOpenInIdeConfigScopeValidator openInIdeConfigScopeValidator,
+        IIDEWindowService ideWindowService)
+    {
+        this.threadHandling = threadHandling;
+        this.logger = logger;
+        this.documentNavigator = documentNavigator;
+        this.issueSpanCalculator = issueSpanCalculator;
+        this.openInIdeConfigScopeValidator = openInIdeConfigScopeValidator;
+        this.ideWindowService = ideWindowService;
+    }
+
+    public void ApplyFixSuggestion(ShowFixSuggestionParams parameters)
+    {
+        if (!ValidateConfiguration(parameters.configurationScopeId, out var configurationScopeRoot, out var failureReason))
+        {
+            logger.WriteLine(FixSuggestionResources.GetConfigScopeRootPathFailed, parameters.configurationScopeId, failureReason);
+            return;
+        }
+        
+        try
+        {
+            logger.WriteLine(FixSuggestionResources.ProcessingRequest, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId);
+
+            var absoluteFilePath = Path.Combine(configurationScopeRoot, parameters.fixSuggestion.fileEdit.idePath);
+            threadHandling.RunOnUIThread(() => ApplySuggestedChanges(absoluteFilePath, parameters.fixSuggestion.fileEdit.changes));
+
+            logger.WriteLine(FixSuggestionResources.DoneProcessingRequest, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId);
+        }
+        catch (Exception exception) when (!ErrorHandler.IsCriticalException(exception))
+        {
+            logger.WriteLine(FixSuggestionResources.ProcessingRequestFailed, parameters.configurationScopeId, parameters.fixSuggestion.suggestionId, exception.Message);
+        }
+    }
+
+    private bool ValidateConfiguration(string configurationScopeId, out string configurationScopeRoot, out string failureReason)
+    {
+        return openInIdeConfigScopeValidator.TryGetConfigurationScopeRoot(configurationScopeId, out configurationScopeRoot, out failureReason);
+    }
+
+    private void ApplySuggestedChanges(string absoluteFilePath, List<ChangesDto> changes)
     {
-        throw new NotImplementedException();
+        ideWindowService.BringToFront();
+        var textView = documentNavigator.Open(absoluteFilePath);
+        for (var i = changes.Count - 1; i >= 0; i--)
```

Comment:
There were two considerations behind this decision:

1. The first change is what triggered the error, so it would make sense to bring the focus to that line.
2. Sometimes, the last change is just a removal of the last empty line; bringing the focus to the first would make more sense than showing the last removed line.

Reply:
The same behavior applies to SLI, it focuses on the first change.